modules/lib/lib-portal/src/main/resources/lib/xp/portal.ts

```diff
@@ -83,15 +83,70 @@ export interface SiteConfig<Config> {
     config: Config;
 }
 
-export interface AssetUrlParams {
+export interface Cookie {
+    value: string;
+    path?: string;
+    domain?: string;
+    maxAge?: number;
+    secure?: boolean;
+    httpOnly?: boolean;
+    sameSite?: 'lax' | 'strict' | 'none' | '';
+}
+
+type LiteralUnion<T extends U, U = string> = T | (U & Record<never, never>);
+
+export type RequestParams = Record<string, string | string[] | undefined>;
+
+export interface Request<Params extends RequestParams = RequestParams,
+    Headers extends Record<string, string | undefined> = Record<string, string | undefined>,
+    Cookies extends Record<string, string | Cookie | undefined> = Record<string, string | Cookie | undefined>,
```

Comment:
Is it correct that a cookie can be `string | Cookie | undefined`? Since this comes from XP, it seems like a risky thing to get a cookie be both `string` and `Cookie`?

Could it be that cookies in `Response` has this shape, but not in `Request`?

Reply:
Request cookies are always `string`. (except for the cases when there is no such cookie)

Reply:
I see, let's make it string/undefined then. But the Response cookie can still be a string/Cookie/undefined, right?